Data_Science/Bank_Loan_Prediction/loan_prediction_system.py

```
@@ -0,0 +1,584 @@
+                precision = precision_score(y_val_split, y_pred, average='weighted', zero_division=0)
+                recall = recall_score(y_val_split, y_pred, average='weighted', zero_division=0)
+                f1 = f1_score(y_val_split, y_pred, average='weighted', zero_division=0)
+                
+                # Handle ROC-AUC for multi-class
+                n_classes = len(np.unique(y_train))
+                if n_classes == 2:
+                    roc_auc = roc_auc_score(y_val_split, y_pred_proba[:, 1])
+                else:
+                    try:
+                        roc_auc = roc_auc_score(y_val_split, y_pred_proba, multi_class='ovr', average='weighted')
+                    except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)